src/Liquid/Filterbank.php

```diff
@@ -144,10 +144,29 @@ public function invoke($name, $value, array $args = array())
 
 		// If we're calling a function
 		if ($class === false) {
-			return call_user_func_array($name, $args);
+			return $this->callUserFuncArrayInTryCatch($name, $args);
 		}
 
 		// Call a class or an instance method
-		return call_user_func_array(array($class, $name), $args);
+		return $this->callUserFuncArrayInTryCatch(array($class, $name), $args);
+	}
+
+	/**
+	 * This is a wrapper for call_user_func_array() with try/catch
+	 * to cast TypeError in LiquidException.
+	 *
+	 * @param callback $function
+	 * @param array $param_arr
+	 *
+	 * @throws \Liquid\LiquidException
+	 * @return mixed
+	 */
+	private function callUserFuncArrayInTryCatch($function, array $param_arr)
+	{
+		try {
+			return call_user_func_array($function, $param_arr);
```

Comment:
I feel like throwing `LiquidException` is beyond what `Filterbank` is supposed to do. Consider we ever want to add a different error handling level. Then we would need to propagate it up to the `Filterbank`, and it will have to know how to deal with different error levels... Well, that's going to be a headache. Best if `Filterbank` would do only what it is doing now, no more.

On the other hand we have `Context::invoke()` which basically proxy-calls the `Filterbank`. You can have this error handling there with the same effect. The `Context` already throws a  `LiquidException` here and there, so there's no questions about it needing to care about different error levels.